docs/tutorials/best-practices/scale-coder.md

```diff
@@ -0,0 +1,309 @@
+# Scale Coder
+
+This best practice guide helps you prepare a low-scale Coder deployment that you can
+scale up to a high-scale deployment as use grows, and keep it operating smoothly with a
+high number of active users and workspaces.
+
+## Observability
+
+Observability is one of the most important aspects to a scalable Coder deployment.
+When you have visibility into performance and usage metrics, you can make informed
+decisions about what changes you should make.
+
+[Monitor your Coder deployment](../../admin/monitoring/index.md) with log output
+and metrics to identify potential bottlenecks before they negatively affect the
+end-user experience and measure the effects of modifications you make to your
+deployment.
+
+- Log output
+  - Capture log output from Loki, CloudWatch logs, and other tools on your Coder Server
+  instances and external provisioner daemons and store them in a searchable log store.
+  - Retain logs for a minimum of thirty days, ideally ninety days.
+  This allows you to look back to see when anomalous behaviors began.
+
+- Metrics
+  - Capture infrastructure metrics like CPU, memory, open files, and network I/O for all
+  Coder Server, external provisioner daemon, workspace proxy, and PostgreSQL instances.
+
+### Capture Coder server metrics with Prometheus
+
+Edit your Helm `values.yaml` to capture metrics from Coder Server and external provisioner daemons with
+[Prometheus](../../admin/integrations/prometheus.md):
+
+1. Enable Prometheus metrics:
+
+   ```yaml
+   CODER_PROMETHEUS_ENABLE=true
+   ```
+
+1. Enable database metrics:
+
+   ```yaml
+   CODER_PROMETHEUS_COLLECT_DB_METRICS=true
+   ```
+
+1. Configure agent stats to avoid large cardinality:
+
+   ```yaml
+   CODER_PROMETHEUS_AGGREGATE_AGENT_STATS_BY=agent_name
+   ```
+
+   - To disable agent stats:
+
+     ```yaml
+     CODER_PROMETHEUS_COLLECT_AGENT_STATS=false
```

Comment:
this and the previous are an either/or situation.  For high scale, you want to either configure the agent stats to avoid large cardinality, or disable them altogether.